respy/tests/random_model.py

```diff
@@ -163,6 +181,14 @@ def generate_random_model(
         "n_periods": np.random.randint(1, bound_constr["max_periods"]),
     }
 
+    if observables is not False:
+        indices = params.loc["observables", :].index.get_level_values(0).to_list()
+        observable_covariates = {
+            x: f'{"_".join(x.split("_")[:-1])} == {x.split("_")[-1]}' for x in indices
```

Comment:
What you want is [rplit](https://docs.python.org/3.7/library/stdtypes.html#str.rsplit)